pkg/monitor/monitor_test.go

```diff
@@ -0,0 +1,245 @@
+
+	for _, n := range nodes.Items {
+		node, err := k8s.Instance().CreateNode(&n, false)
+		require.NoError(t, err, "failed to create fake node")
+		require.NotNil(t, node, "got nil node from create node api")
+	}
+
+	provNodes := []int{0, 1}
+	err = driver.CreateCluster(5, nodes)
+	require.NoError(t, err, "Error creating cluster")
+
+	err = driver.ProvisionVolume(driverVolumeName, provNodes, 1)
+	require.NoError(t, err, "Error provisioning volume")
+}
+
+func testUnknownDriverPod(t *testing.T) {
+	pod := newPod("driverPod", []string{driverVolumeName})
+	testLostPod(t, pod, true, true, false)
+}
+
+func testUnknownOtherDriverPod(t *testing.T) {
+	pod := newPod("otherDriverPod", nil)
+	podVolume := v1.Volume{}
+	podVolume.PersistentVolumeClaim = &v1.PersistentVolumeClaimVolumeSource{
+		ClaimName: "noDriverPVC",
+	}
+	pod.Spec.Volumes = append(pod.Spec.Volumes, podVolume)
+
+	testLostPod(t, pod, false, true, false)
+}
+
+func testEvictedOtherDriverPod(t *testing.T) {
+	pod := newPod("otherDriverPod", nil)
+	podVolume := v1.Volume{}
+	podVolume.PersistentVolumeClaim = &v1.PersistentVolumeClaimVolumeSource{
+		ClaimName: "noDriverPVC",
+	}
+	pod.Spec.Volumes = append(pod.Spec.Volumes, podVolume)
+
+	testLostPod(t, pod, false, false, true)
+}
+
+func testEvictedDriverPod(t *testing.T) {
+	pod := newPod("driverPod", []string{driverVolumeName})
+	testLostPod(t, pod, true, false, true)
+}
+
+func testLostPod(
+	t *testing.T,
+	pod *v1.Pod,
+	driverPod bool,
+	testUnknownPod bool,
+	testTaintBasedEviction bool,
+) {
+	pod, err := k8s.Instance().CreatePod(pod, false)
+	require.NoError(t, err, "failed to create pod")
+	require.NotNil(t, pod, "got nil pod back from create pod")
+
+	pod, err = k8s.Instance().GetPodByName(pod.Name, "")
+	require.NoError(t, err, "failed to get pod from fake API")
+	require.NotNil(t, pod, "got nil pod back from get pod")
+
+	logrus.Infof("[debug] pod: %v", pod)
+
+	info, err := driver.GetPodVolumes(&pod.Spec, "")
+	if driverPod {
+		require.NoError(t, err, "failed to get pod from fake API")
+		require.NotNil(t, pod, "got nil pod back from get pod")
+	} else {
+		require.Error(t, err, "expectd error when getting pod volumes from mock driver")
+		require.Nil(t, info, "expected empty pod volumes from driver")
+	}
+
+	if testUnknownPod {
+		// make pod unknown
+		pod.Status = v1.PodStatus{
+			Reason: node.NodeUnreachablePodReason,
+		}
+	}
+
+	if testTaintBasedEviction {
+		pod.ObjectMeta.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+		node, err := k8s.Instance().GetNodeByName(nodeForPod)
+		require.NoError(t, err, "failed to get node for pod")
+		/*node.Spec.Taints = make([]v1.Taint, 1)
+		node.Spec.Taints[0] = v1.Taint{
+			Key:    "key",
+			Effect: v1.TaintEffectNoExecute,
+		}*/
+		node.Spec.Taints = []v1.Taint{
+			v1.Taint{
+				Key:    algorithm.TaintNodeUnreachable,
+				Effect: v1.TaintEffectNoExecute,
+			},
+		}
+
+		node, err = k8s.Instance().CreateNode(node, true)
```

Comment:
UpdateNode?

Reply:
Also, don't assign `node`, this will add an error in GoReport for ineffectual assignment since it's not being used after this

Reply:
done